site/guides/effective-go-krm-function.md

```diff
@@ -0,0 +1,212 @@
+# Effective GO KRM functions
+
+This guide gives tips to effectively write a KRM function. 
+
+This guide is for advanced kpt function users who find the [catalog.kpt.dev] cannot fulfil their needs 
+and want to design their own KRM functions. 
+
+Suggest reading [Developing in Go] first
+
+## Prerequisites
+
+- [Install kpt]
+- [Install Docker]
+
+## Setup 
+
+<!--- TODO: Use scaffolding to generate the get-started package --->
+
+We start from a "get-started-runner" package which contains a `main.go` file with some scaffolding code.
+
+
+```shell
+# Set your KRM function name.
+export FUNCTION_NAME=<YOUR FUNCTION NAME>
+export GOPATH=$(go env GOPATH)
+export FUNCTION_PATH=github.com/<YOUR USERNAME>
+
+# Create and direct to your Go working directory
+mkdir -p $GOPATH/src/${FUNCTION_PATH} && cd $GOPATH/src/${FUNCTION_PATH}
+
+# Get the "get-started" package.
+kpt pkg get https://github.com/GoogleContainerTools/kpt-functions-sdk.git/go/get-started-runner@master ${FUNCTION_NAME}
```

Comment:
Assuming myself as a Go developer, I will find this step to be awkward :) and I will also be surprised to see files such as `Kptfile` etc. It might get confusing. Can we just use `git clone` or something to fetch the skeleton code ?